docs/kratos/social-signin/05_generic.mdx

```diff
@@ -68,3 +68,79 @@ import JsonnetWarning from '../../_common/jsonnetwarning.mdx'
 
 <JsonnetWarning format="Jsonnet code snippets" use="data mapping" />
 ```
+
+## Configuration using the Ory CLI
+
+Follow these steps to add a generic provider as a social sign-in provider to your Ory Cloud project using the Ory CLI:
+
+1. Create a client at your generic provider to get a Client ID and Client Secret.
+2. Create a [Jsonnet code snippet](#data-mapping) to map the desired claims to the Ory Identity schema.
+3. Encode the Jsonnet snippet with [Base64](https://www.base64encode.org/) or host it under an URL accessible to Ory Cloud.
+4. Download the Identity Service config from your Ory Cloud project and save it to a file:
+
+   ```shell
+   ## list all available projects
+   ory list projects
+
+   ## store config
+   ory get identity-config <project-id> --format yaml > identity-config.yaml
+   ```
+
+5. Add the social sign-in provider configuration to the downloaded config. Add the Jsonnet snippet with mappings as a Base64
+   string or provide an URL to the file.
+
+   ```yaml
+   selfservice:
+     methods:
+       oidc:
+         config:
+           providers:
+             - id: generic # this is `<provider-id>` in the Authorization callback URL. DO NOT CHANGE IT ONCE SET!
+               provider: generic
+               client_id: .... # Replace this with the Client ID
+               client_secret: .... # Replace this with the Client secret
+               mapper_url: "base64://<YOUR_BASE64_ENCODED_JSONNET_HERE>"
+               # Alternatively, use an URL:
+               # mapper_url: https://storage.googleapis.com/abc-cde-prd/9cac9717f007808bf17f22ce7f4295c739604b183f05ac4afb4
+               scope:
+                 - email
+               # supported scopes can be found in your providers dev docs
+         enabled: true
+   ```
+
+6. Update the Ory Identity Service (Ory Kratos) configuration using the file you worked with:
+
+   ```shell
+   ory update identity-config <project-id> --file updated_config.yaml
+   ```
+
+## Configuration for self-hosted instances
+
+Follow these steps to add a generic provider as a social sign-in provider when self-hosting Ory Kratos:
+
+1. Create a client at your generic provider to get a Client ID and Client Secret.
+2. Set the redirect URI to URL that follows this pattern:
+
+   ```shell
+   http(s)://<domain-of-ory-kratos>:<public-port>/self-service/methods/oidc/callback/<social-signin-provider-id>
+   ```
+
+3. Create a [Jsonnet code snippet](#data-mapping) to map the desired claims to the Ory Identity schema.
+4. Encode the Jsonnet snippet with [Base64](https://www.base64encode.org/) or store it in a location available to your Ory Kratos
+   instance.
+5. Add the [social sign-in provider configuration](#configuration-with-ory-cli) to the Ory Kratos configuration. Add the Jsonnet
+   snippet with mappings as a Base64 string or provide a path or an URL of the file.
+
+```mdx-code-block
+import ConfigAsEnv from '../_common/config_as_env.mdx'
+
+<ConfigAsEnv />
+```
```

Comment:
The console UI has the users provide an issuer URL when adding a generic provider. This isn't the case for custom integrations like GH or Apple. The process description and the relevant code snippets should include this information. 

I added a generic provider through the console UI and here's how it was added to the config.yaml: 

```
- client_id: "2352545225"
          client_secret: "2433443234"
          id: jHKCZmrsWhcdbXj_
          issuer_url: https://accounts.google.com
          label: www
          mapper_url: REDACTED
          provider: generic
          scope:
          - s
      enabled: true
```

Reply:
Ah good point, I will add the issuer URL, otherwise seems to be the same, 
just not sure what "label" does. I dont think its necessary though.